cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -128,26 +128,32 @@ pub enum Error {
 
 impl Cmd {
     pub async fn run(&self, global_args: &global::Args) -> Result<(), Error> {
-        let res = self
-            .run_against_rpc_server(Some(global_args), None)
-            .await?
-            .to_envelope();
+        let res = self.run_against_rpc_server(Some(global_args), None).await;
         match res {
-            TxnEnvelopeResult::TxnEnvelope(tx) => println!("{}", tx.to_xdr_base64(Limits::none())?),
-            TxnEnvelopeResult::Res(contract) => {
-                let network = self.config.get_network()?;
-
-                if let Some(alias) = self.alias.clone() {
-                    self.config.locator.save_contract_id(
-                        &network.network_passphrase,
-                        &contract,
-                        &alias,
-                    )?;
+            Ok(res) => match res.to_envelope() {
+                TxnEnvelopeResult::TxnEnvelope(tx) => {
+                    println!("{}", tx.to_xdr_base64(Limits::none())?);
                 }
-
-                println!("{contract}");
+                TxnEnvelopeResult::Res(contract) => {
+                    let network = self.config.get_network()?;
+
+                    if let Some(alias) = self.alias.clone() {
+                        self.config.locator.save_contract_id(
+                            &network.network_passphrase,
+                            &contract,
+                            &alias,
+                        )?;
+                    }
+
+                    println!("{contract}");
+                }
+            },
+            Err(ArgParse(HelpMessage(help))) => {
+                println!("{help}");
```

Comment:
I think we need to do this at the top level, rather than inside the arg parsing, otherwise the exit codes will be incorrect when generating the help when no `--help` is passed. It's a small thing, but also I believe a small fix to keep passing the error as is up the chain, and at the top level doing as you do here.

Reply:
Do you mean we should handle HelpMessage error at the very [top of the chain?](https://github.com/stellar/stellar-cli/blob/61550ec2157e5550dfaa967fa20c2ca9e2de4df2/cmd/soroban-cli/src/cli.rs#L87)